gen/briefinggen.py

```diff
@@ -24,19 +31,144 @@ class CommInfo:
     freq: RadioFrequency
 
 
+class FrontLineInfo:
+    def __init__(self, front_line: FrontLine):
+        self.front_line: FrontLine = front_line
+        self.player_base: ControlPoint = front_line.control_point_a
+        self.enemy_base: ControlPoint = front_line.control_point_b
+        self.player_zero: bool = self.player_base.base.total_armor == 0
+        self.enemy_zero: bool = self.enemy_base.base.total_armor == 0
+        self.advantage: bool = self.player_base.base.total_armor > self.enemy_base.base.total_armor
+        self.stance: CombatStance = self.player_base.stances[self.enemy_base.id]
+
+    @property
+    def _random_frontline_sentence(self) -> str:
+        '''Random sentences for start of situation briefing'''
+        templates = [
+            f"There are combats between {self.player_base.name} and {self.enemy_base.name}. ",
+            f"The war on the ground is still going on between {self.player_base.name} and {self.enemy_base.name}. ",
+            f"Our ground forces in {self.player_base.name} are opposed to enemy forces based in {self.enemy_base.name}. ",
+            f"Our forces from {self.player_base.name} are fighting enemies based in {self.enemy_base.name}. ",
+            f"There is an active frontline between {self.player_base.name} and {self.enemy_base.name}. ",
+        ]
+        return random.choice(templates)
+
+    @property
+    def _zero_units_sentence(self) -> str:
+        '''Situation description if either side has zero units on a frontline'''
+        if self.player_zero:
+            return ("We do not have a single vehicle available to hold our position.  The situation is critical, "
+                    f"and we will lose ground inevitably betweeen {self.player_base.name} and {self.enemy_base.name}")
+        elif self.enemy_zero:
+            return ("The enemy forces have been crushed, we will be able to make significant progress"
+                    f" toward {self.enemy_base.name}.")
+        return ''
```

Comment:
A minor thing, but try to stay consistent with quote use. The code base is mostly double quotes atm (same goes for docstrings). Exception is is you need to do `"foo said \"bar\""`, using `'foo said "bar"'` is fine then.

https://www.python.org/dev/peps/pep-0008/#string-quotes

Reply:
I heavily rely on Black in my day to day to fix all my single quotes, but it makes a big git mess in this project.  I'll get my formatter configured, that's far more likely to stick than the counting on my habits changing 🙃

Reply:
Changed all single quotes I added to double